src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -166,6 +167,13 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     m_dev_list = ov::DeviceIDParser::get_hetero_devices(dev_list_str);
     m_meta_devices = ov::npuw::get_properties_per_device(plugin, dev_list_str, m_non_npuw_props);
 
+    if (m_name.find(NPUW_NAME_IDENTIFIER) != m_name.npos) {
+        LOG_DEBUG("CompiledModel is being deserialized, skipping the full constructor flow...");
+        // Drop serialized from the name
+        m_name = m_name.substr(std::string(NPUW_NAME_IDENTIFIER).size());
+        return;
+    }
```

Comment:
What is this? Why do we need this? Why?

Reply:
In constructor of `CompiledModel` and `LLMCompiledModel` I somehow need to identify that it's being called from `deserialize()`. So I need some mechanism to distinguish default constructor behavior from a dummy deserialized one

Reply:
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/plugin/compiled_model.cpp#L66